src/bluesky/bundlers.py

```diff
@@ -641,54 +632,26 @@ async def kickoff(self, msg):
         """
         self._uncollected.add(msg.obj)
 
-    # Could we have a look at changing this now? we use one of each use can in
-    # seperate places so it could be two seperate methods for each dictionary type.
-    def _maybe_format_datakeys_with_stream_name(
-        self,
-        describe_collect_dict: Union[Dict[str, DataKey], Dict[str, Dict[str, DataKey]]],
-        message_stream_name: Optional[str] = None,
-    ) -> List[Tuple[str, Dict[str, DataKey]]]:
-        """
-        Check if the dictionary returned by describe collect is a dict
-            `{str: DataKey}` or a `{str: {str: DataKey}}`.
-        If a `message_stream_name` is passed then return a singleton list of the form of
-            `{message_stream_name: describe_collect_dict}.items()`.
-        If the `message_stream_name` is None then return the `describe_collect_dict.items()`.
-        """
-
-        def has_str_source(d: dict):
-            return isinstance(d, dict) and isinstance(d.get("source", None), str)
-
-        if describe_collect_dict:
-            first_value = list(describe_collect_dict.values())[0]
-            if has_str_source(first_value):
-                # We have Dict[str, DataKey], so return just this
-                # If stream name not given then default to "primary"
-                return [(message_stream_name or "primary", describe_collect_dict)]
-            elif all(has_str_source(v) for v in first_value.values()):
-                # We have Dict[str, Dict[str, DataKey]] so return its items
-                if message_stream_name and list(describe_collect_dict) != [message_stream_name]:
-                    # The collect contained a name and describe_collect returned a Dict[str, Dict[str, DataKey]],
-                    # this is only acceptable if the only key in the parent dict is message_stream_name
-
-                    raise RuntimeError(
-                        f"Expected a single stream {message_stream_name!r}, got {describe_collect_dict}"
-                    )
-                return list(describe_collect_dict.items())
-            else:
-                raise RuntimeError(
-                    f"Invalid describe_collect return: {describe_collect_dict} when collect "
-                    f"was called on {message_stream_name}"
-                )
-        else:
-            # Empty dict, could be either but we don't care
-            return []
+    async def _cache_describe_collect(self, obj: Collectable, stream_name: Optional[str] = None):
+        def is_data_key(obj: Any) -> bool:
+            return isinstance(obj, dict) and {"dtype", "shape", "source"}.issubset(frozenset(obj.keys()))
 
-    async def _cache_describe_collect(self, obj: Collectable):
         "Read the object's describe and cache it."
         obj = check_supports(obj, Collectable)
         c = await maybe_await(obj.describe_collect())
```

Comment:
Is `c` a standard name for collected objects? Would it be possible to use a more descriptive variable name here?